### PR TITLE
json-export: Support multi-map like objects

### PR DESCRIPTION
**Pull request by johannes-wolf**

### CODE REVIEW

**Comment by johannes-wolf** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=Klebert-Engineering_simfil&pullRequest=92) **Quality Gate failed**
> 
> Failed conditions  [C Maintainability Rating on New Code](https://sonarcloud.io/dashboard?id=Klebert-Engineering_simfil&pullRequest=92) (required ≥ A)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=Klebert-Engineering_simfil&pullRequest=92)
> 
>  Catch issues before they fail your Quality Gate with our IDE extension  [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=pull-request)

Chill!

**Comment by josephbirkner** [comment]
I'm wondering if we could do the following:
* If we notice that the object is a multi-map due to a duplicate key-value-pair, then turn *every* member into a list, and add a `_multimap: true` field.
* This would allow us to reconstruct the object in `fromJson`.
* Alternatively, the `_multimap` member could list the member names that have multiple values.

File: test/complex.cpp [hunk 1]
```diff
@@ -90,6 +91,28 @@ TEST_CASE("Runtime Error", "[yaml.complex.runtime-error]") {
     REQUIRE_THROWS(joined_result("not *.nonexisting == 0"));   /* Invalid operands int and bool */
 }
 
+TEST_CASE("Multimap JSON", "[yaml.multimap.serialization]") {
```
Comment (josephbirkner) [inline test/complex.cpp:94]: yaml?

**Review by josephbirkner — APPROVED** [review]
Awesome, thanks :smile:

**Comment by josephbirkner** [comment]
I think after this change, we can deploy a new simfil version to conan-center-index as well.

**Pull request merged**